### PR TITLE
Contact token client support

### PR DESCRIPTION
**Pull request by heronhaye**
- [x] try this out while watching ratelimits to make sure tokens work

### CODE REVIEW

**Comment by heronhaye** [comment]
I'll rebuild but chat test has flaked twice in a row

**Comment by heronhaye** [comment]
Ping

File: go/service/contacts.go [hunk 1]
```diff
@@ -20,11 +20,18 @@ type bulkLookupContactsProvider struct{}
 
 var _ contacts.ContactsProvider = (*bulkLookupContactsProvider)(nil)
 
+func (c *bulkLookupContactsProvider) LookupAllWithToken(mctx libkb.MetaContext, emails []keybase1.EmailAddress,
+	numbers []keybase1.RawPhoneNumber, userRegion keybase1.RegionCode, token contacts.Token) (contacts.ContactLookupResults, error) {
+	defer mctx.TraceTimed(fmt.Sprintf("bulkLookupContactsProvider#LookupAllWithToken(len=%d)", len(emails)+len(numbers)),
+		func() error { return nil })()
+	return contacts.BulkLookupContacts(mctx, emails, numbers, userRegion, token)
```
Comment (zapu) [inline go/service/contacts.go]: I think the tokens are implementation-specific enough that we can just handle them inside of `contacts.BulkLookupContacts` and not "leak" outside of that package. What do you think?

Comment (zapu) [reply go/service/contacts.go]: hm, that can't be exactly that... maybe that's necessary after all, too bad

File: go/contacts/cache.go [hunk 2]
```diff
@@ -217,8 +223,10 @@ func (c *CachedContactsProvider) LookupAll(mctx libkb.MetaContext, emails []keyb
 	mctx.Debug("After checking cache, %d emails and %d numbers left to be looked up", len(remainingEmails), len(remainingNumbers))
 
 	if len(remainingEmails)+len(remainingNumbers) > 0 {
-		apiRes, err := c.Provider.LookupAll(mctx, remainingEmails, remainingNumbers, userRegion)
+		apiRes, err := c.Provider.LookupAllWithToken(mctx, remainingEmails, remainingNumbers, userRegion, conCache.Token)
 		if err == nil {
+			conCache.Token = apiRes.Token
```
Comment (zapu) [inline go/contacts/cache.go]: looks like the adhoc tofu searches for just one phonenumber/email will always overwrite your big tokens for contact lists

Comment (heronhaye) [reply go/contacts/cache.go]: I don't think so, the server sends back a token for everything in the token as well as what was requested

Comment (heronhaye) [reply go/contacts/cache.go]: sorry I could be wrong about that...but if the server did do that, it'd be ok?

File: go/service/contacts.go (hunk 1 above)
Comment (heronhaye) [reply go/service/contacts.go]: yeah i tried to make it not leak but found it hard with them both satisfying the same interface

**Comment by heronhaye** [comment]
merging over lint errors that I didn't cause

**Pull request merged**